### ISSUE TITLE
Form-Field `input` and `label` DOM order issue

### ISSUE THREAD

**Issue by sjbeatle** [angular/components#18989]
#### Bug, feature request, or proposal:
According to WCAG [H44: Using label elements to associate text labels with form controls](https://www.w3.org/TR/WCAG20-TECHS/H44.html)
> Check that there is a label element that identifies the purpose of the control **before** the input, textarea, or select element

In the Material [form-field](https://github.com/angular/components/blob/master/src/material/form-field/form-field.html#L24) component, the `label` is **after** the `input`

#### What is the expected behavior?
The Material [form-field](https://github.com/angular/components/blob/master/src/material/form-field/form-field.html#L24) renders the `label` **before** the `input` in the DOM according the the WCAG.

#### What is the current behavior?
The form field component renders the `label` **after** the `input` in the DOM. Accessibility tests are failing because of this, and it is likely why JAWS and NVDA weren't reading the labels. IE, `aria-owns` is likely unneeded if the DOM order were corrected.

#### What are the steps to reproduce?
1. Use a [form-field](https://github.com/angular/components/blob/master/src/material/form-field/form-field.html) component in your project

#### What is the use-case or motivation for changing an existing behavior?
Accessibility and WCAG compliance. You will likely be able to remove the superfluous `aria-owns` attribute as a result, as well.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
All of them, I believe... At least 5.X and up.

#### Is there anything else we should know?
That is all. Thank you!

**Comment by devversion** [comment]
That's a great suggestion. Thanks! 

The MDC-based form-field has the label before the projected content (e.g. the `<input>` element) in the filled variant. In the outline appearance, the label is also before, but hierarchically higher.

**Comment by annieyw** [comment]
Confirmed that MDC outline appearance text field has label before input

**Comment by Erbenos** [comment]
There seems to have been a PR opened for this, but it wasn't merged for some reason #7507.

**Comment by amysorto** [comment]
The MDC based form field (which is now available in v15) has the correct order.
